commit 395cbb9a52bad9d15d2220c445b067c1cde9f69f
Author: Jiri Olsa <jolsa@kernel.org>
Date:   Wed Dec 26 12:21:21 2018 +0100

    perf python: Do not force closing original perf descriptor in evlist.get_pollfd()
    
    [ Upstream commit a389aece97938966616ce0336466b98b0351ef10 ]
    
    Ondřej reported that when compiled with python3, the python extension
    regresses in evlist.get_pollfd function behaviour.
    
    The evlist.get_pollfd function creates file objects from evlist's fds
    and returns them in a list. The python3 version also sets them to 'close
    the original descriptor' when the object dies (is closed), by passing
    True via the 'closefd' arg in the PyFile_FromFd call.
    
    The python's closefd doc says:
    
      If closefd is False, the underlying file descriptor will be kept open
      when the file is closed.
    
    That's why the following line in python3 closes all evlist fds:
    
      evlist.get_pollfd()
    
    the returned list is immediately destroyed and that takes down the
    original events fds.
    
    Passing closefd as False to PyFile_FromFd to fix this.
    
    Reported-by: Ondřej Lysoněk <olysonek@redhat.com>
    Signed-off-by: Jiri Olsa <jolsa@kernel.org>
    Cc: Alexander Shishkin <alexander.shishkin@linux.intel.com>
    Cc: Jaroslav Škarvada <jskarvad@redhat.com>
    Cc: Namhyung Kim <namhyung@kernel.org>
    Cc: Peter Zijlstra <peterz@infradead.org>
    Fixes: 66dfdff03d19 ("perf tools: Add Python 3 support")
    Link: http://lkml.kernel.org/r/20181226112121.5285-1-jolsa@kernel.org
    Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/tools/perf/util/python.c b/tools/perf/util/python.c
index ce501ba14b08..69f5f6142dcf 100644
--- a/tools/perf/util/python.c
+++ b/tools/perf/util/python.c
@@ -939,7 +939,8 @@ static PyObject *pyrf_evlist__get_pollfd(struct pyrf_evlist *pevlist,
 
 		file = PyFile_FromFile(fp, "perf", "r", NULL);
 #else
-		file = PyFile_FromFd(evlist->pollfd.entries[i].fd, "perf", "r", -1, NULL, NULL, NULL, 1);
+		file = PyFile_FromFd(evlist->pollfd.entries[i].fd, "perf", "r", -1,
+				     NULL, NULL, NULL, 0);
 #endif
 		if (file == NULL)
 			goto free_list;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1215_395cbb9a52bad9d15d2220c445b067c1cde9f69f.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1215_395cbb9a52bad9d15d2220c445b067c1cde9f69f.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
